code/centauri/ibc/modules/README.md

```diff
@@ -1,47 +1,214 @@
 # IBC module
 
-[![Crate][crate-image]][crate-link]
-[![Docs][docs-image]][docs-link]
-[![Build Status][build-image]][build-link]
-[![End to End testing][e2e-image]][e2e-link]
 [![Apache 2.0 Licensed][license-image]][license-link]
-![Rust Stable][rustc-image]
-![Rust 1.51+][rustc-version]
 
+Implementation of the Inter-Blockchain Communication Protocol ([IBC]) in rust.
 
-See the [ibc-rs] repo root for more detailed information on how this crate can be used.
+## Project Structure
 
+**Core** contains the traits and handlers that enable the ibc protocol.  
+**Applications** contains sub protocols that are built off the core ibc.  
+**Mock** contains implementations of the core ibc protocol for testing purposes.  
 
-Implementation of the Inter-Blockchain Communication Protocol ([IBC]) module.
+## Architecture
 
-## Documentation
+The design of this crate is geared towards making it as performant as possible and reducing runtime overhead to the bear minimum.  
```

Comment:
> The design of this crate is geared towards making it as performant as possible and reducing runtime overhead to the bear minimum.
This line can be removed :) Sure it was coded like that.

Reply:
Yeah, I think static dispatch is pretty standard and not considered head-over-heels for performance. If instead we had used dynamic dispatch, unsafe or asm everywhere, an explanation and warning would make sense. But the core::vec docs don't start talking about the noble goal of static dispatch either ;)